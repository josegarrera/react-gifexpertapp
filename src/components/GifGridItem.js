import React from 'react';
import PropTypes from 'prop-types';

function GifGridItem({ id, title, url }) {

    return (
        <div className='card animate__animated animate__fadeInUp' >
            <img key={id} src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
};

GifGridItem.propTypes = {
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  };

export default GifGridItem
