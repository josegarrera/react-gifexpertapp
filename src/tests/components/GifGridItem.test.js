import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';



    describe('Tests on <GifGridItem />', () => {

        const id = 19;
        const title = 'A title';
        const url = 'https://google.com/something.jpg';
        const wrapper = shallow(<GifGridItem id={id} title={title} url={url} />);

        test('Should render component', () => {
            expect(wrapper).toMatchSnapshot();
        });

        test('Should have a paragraph with the title', () => {
            const p = wrapper.find('p');
            expect(p.text().trim()).toBe(title);    
        });

        test('Should have an image with src equals to url and alt equals to title', () => {
            const img = wrapper.find('img');
            expect(img.props().src).toBe(url);
            expect(img.props().alt).toBe(title);
        });

        test('Should have the className animate__fadeIn', () => {
            const div = wrapper.find('div');
            expect(div.props().className).toContain('animate__fadeIn')
            // otra forma
            // expect(div.props().className.includes('animate__fadeIn')).toBe(true);
        });
    });