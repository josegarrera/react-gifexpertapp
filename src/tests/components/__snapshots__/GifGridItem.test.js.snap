// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem /> Should render component 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="A title"
    key="19"
    src="https://google.com/something.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
